src/linear_algebra/basic.lean

```diff
@@ -2102,10 +2102,47 @@ end
 
 end field
 
+/-- Given `f : M₁ ≃ₗ[R] M₂` and `U` a submodule of `M₂`, `f.comap U` is the
+induced `linear_equiv` from `U.comap f.to_linear_map` to `U`. -/
+def comap [ring R] [add_comm_group M] [module R M]
+  [add_comm_group M₂] [module R M₂] (f : M ≃ₗ[R] M₂) (U : submodule R M₂) :
+  U.comap f.to_linear_map ≃ₗ[R] U := f.of_submodules _ _
+begin
+  ext x,
+  simp_rw [submodule.mem_map, submodule.mem_comap],
+  split; intro hx,
+  { rcases hx with ⟨y, hy, rfl⟩,
+    exact hy },
+  { refine ⟨f.inv_fun x, _⟩,
+    simpa }
+end
+
 end linear_equiv
 
 namespace submodule
 
+/-- Given `p` a submodule of the module `M` and `q` a submodule of `p`, `p.equiv_subtype_map q`
+is the natural `linear_equiv` between `q` and `q.map p.subtype`. -/
+noncomputable def equiv_subtype_map [ring R] [add_comm_group M] [module R M]
```

Comment:
Same as above: does `@[simps]` work and do you have any unfolds of this definition?
```suggestion
@[simps]
noncomputable def equiv_subtype_map [ring R] [add_comm_group M] [module R M]
```